src/answer/answer.service.ts

```diff
@@ -1,147 +1,259 @@
+        await this.answerRepository.save(createdAnswer);
+        const userDto = await this.usersService.getUserDtoById(userId);
+        return {
+            id: createdAnswer.id,
+            question_id: createdAnswer.question_Id,
+            content: createdAnswer.content,
+            author: userDto,
+            created_at: createdAnswer.createdAt.getTime(),
+            updated_at: createdAnswer.updatedAt.getTime(),
+            // agree_type: 0,
+        };
     }
 
-    return queryBuilder.getMany();
-  }
+    async getQuestionAnswers(
+        questionId: number | undefined, 
+        page_start: number | undefined, 
+        page_size: number
+    ): Promise<[AnswerDto[], PageRespondDto]> {
+        let queryBuilder = this.answerRepository
+        .createQueryBuilder('answer')
+        .where('answer.questionId = :questionId', { questionId })
+        .orderBy('answer.createdAt');
+        let prevPage = undefined, currPage = undefined;
+        if(!page_start){
+            //排序？
+            currPage = await queryBuilder.limit(page_size + 1).getMany();
+            const currDto = await Promise.all(currPage.map(async (entity) => {
+                const userId = await this.getUserByAnswer(entity.id);
+                return this.getAnswerById(userId, questionId, entity.id);
+            }));
+            return PageHelper.PageStart(
+                currDto,
+                page_size,
+                (answer) => answer.id,
+            );
+        }
+        else{
+            const start = await this.answerRepository.findOneBy({id: page_start});
+            if (!start){
+                throw new AnswerNotFoundError(page_start);
+            }
+            const queryBuilderCopy = queryBuilder.clone();
+            prevPage = await queryBuilder
+                .orderBy('id', 'ASC')
+                .andWhere('id > :page_start', {page_start})
+                .limit(page_size)
+                .getMany();
+            currPage = await queryBuilderCopy
+                .orderBy('id', 'DESC')
+                .andWhere('id <= :page_start', {page_start})
+                .limit(page_size + 1)
+                .getMany();
 
-  async updateAnswer(
-    id: number,
-    updateAnswerDto: UpdateAnswerDto,
-  ): Promise<Answer> {
-    const answer = await this.answerRepository.findOne({ where: { id } });
+            const currDto = await Promise.all(currPage.map(async (entity) => {
+                const userId = await this.getUserByAnswer(entity.id);
+                return this.getAnswerById(userId, questionId, entity.id);
+            }));
+            return PageHelper.PageMiddle(
+                prevPage,
+                currDto,
+                page_size,
+                (answer) => answer.id,
+                (answer) => answer.id,
+            );
 
-    if (!answer) {
-      throw new AnswerNotFoundError(id);
+        }
+        
+    }
+    
+    async getUserByAnswer(answerId: number){
```

Comment:
If this method is only used by internal methods, then it should be marked as `private`.
And I think the better way here is to add a lambda, and finding without checking, since checking entities just fetched from database is somewhat useless.